src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultDocumentSnapshot.cs

```diff
@@ -88,36 +88,42 @@ public override bool TryGetGeneratedOutput(out RazorCodeDocument result)
         {
             if (State.IsGeneratedOutputResultAvailable)
             {
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
                 result = State.GetGeneratedOutputAndVersionAsync(ProjectInternal, this).Result.output;
```

Comment:
Sucks that we got cornered into doing `.Result`, would it be feasible to make this `TryGetGeneratedOutputAsync`?

Reply:
Unfortunately not because this call path is intended to be synchronous.